### PR TITLE
Module renamed to prince_cr

### LINKED ISSUE #32: Rename module to PriNCe and move tests into prince directory
I think one last change that would break the examples and should be done before reaching milestone is the renaming of the module to PriNCe.
On the same note I recommend/prefer pulling the tests into the prince package folder. This has some more academic implications, like prevention from triggering local imports that can the correctness of paths etc. I have seen that many packages do it like this and also have done it successfully in MCEq. Pytest expects this and will find these tests by default.

### PR DESCRIPTION
**Pull request by afedynitch**
As discussed this will close #32

### CODE REVIEW

**Comment by afedynitch** [comment]
Please wait. I will silence a waning and add another commit to this PR.

**Comment by afedynitch** [comment]
Done

File: prince/_deprecated/util.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import numpy as np
```
Comment (joheinze) [inline prince/_deprecated/util.py:1]: Did you forget to remove this module?

File: prince_cr/_deprecated/util.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import numpy as np
```
Comment (joheinze) [inline prince_cr/_deprecated/util.py:1]: Ok, yes I think that git did not recognize this file was moved.
I will fix that

File: prince_cr/solvers/propagation.py [hunk 3]
```diff
@@ -113,13 +113,15 @@ def get_solution_group(self, nco_ids, epow=3, egrid=None):
...
+            mask = curr_spec > 0.
+            if np.count_nonzero(mask) > 0:
+                res = np.exp(
+                    np.interp(
+                        np.log(com_egrid),
+                        np.log(curr_egrid[mask]),
+                        np.log(curr_spec[mask]),
+                        left=np.nan,
+                        right=np.nan))
             spectra[idx] = np.nan_to_num(res)
```
Comment (joheinze) [inline prince_cr/solvers/propagation.py:125]: This change is to fix the numpy overflow warnings, right? But what happens if curr_spec is all `0`? Then the if statement it skipped and l.125 will throw an error. 

I think this should look like this (have not tested):
```suggestion
            mask = curr_spec > 0.
            if np.count_nonzero(mask) > 0:
                res = np.exp(
                    np.interp(
                        np.log(com_egrid),
                        np.log(curr_egrid[mask]),
                        np.log(curr_spec[mask]),
                        left=np.nan,
                        right=np.nan))
            else:
                res = np.zeros_like(com_egrid)
            spectra[idx] = np.nan_to_num(res)
```

File: prince_cr/solvers/propagation.py [hunk 4]
```diff
@@ -448,29 +452,34 @@ def eqn_deriv_mkl(self, z, state, *args):
...
         else:
             res = csrmv(1., self.jacobian, state, 0., res)
-            
 
         if self.enable_injection_jacobian:
-            res += self.injection(1., z)[:, np.newaxis]
+            try:
+                res += self.injection(1., z)[:, np.newaxis]
+            except ValueError:
+                res += self.injection(1., z)
```
Comment (joheinze) [inline prince_cr/solvers/propagation.py:465]: This fixes the problem with the Euler solver, right?

This exception here is a little confusing. Maybe we should rather adjust this in the Euler solver class.

Or atleast leave a comment:
```suggestion
            # catch error due to different shape of res in UHECRPropagationSolverEULER
            try:
                res += self.injection(1., z)[:, np.newaxis]
            except ValueError:
                res += self.injection(1., z)
```

File: prince_cr/solvers/propagation.py [hunk 5]
```diff
@@ -448,29 +452,34 @@ def eqn_deriv_mkl(self, z, state, *args):
...
         if self.enable_partial_diff_jacobian:
             conloss = np.zeros_like(self.adia_loss_rates_grid.energy_vector)
             if self.enable_adiabatic_losses:
                 conloss += self.adia_loss_rates_grid.loss_vector(z)
             if self.enable_pairprod_losses:
                 conloss += self.pair_loss_rates_grid.loss_vector(z)
+            try:
+                loss_state = conloss[:, np.newaxis] * state
+            except ValueError:
+                loss_state = conloss * state
```
Comment (joheinze) [inline prince_cr/solvers/propagation.py:476]: Same as last comment, maybe rather fix in Euler solver
```suggestion
            # catch error due to different shape of res in UHECRPropagationSolverEULER
            try:
                loss_state = conloss[:, np.newaxis] * state
            except ValueError:
                loss_state = conloss * state
```

File: prince/_deprecated/util.py (hunk 1 above)
Comment (afedynitch) [reply prince/_deprecated/util.py:1]: Yes

**Pull request closed without merging**